src/AutoRest.CSharp/Common/Output/Models/Types/ExpressionTypeProvider.cs

```diff
@@ -32,6 +30,11 @@ internal static IEnumerable<ExpressionTypeProvider> GetHelperProviders()
                 yield return ClientPipelineExtensionsProvider.Instance;
                 yield return ClientUriBuilderProvider.Instance;
             }
+            if (Configuration.IsBranded)
```

Comment:
I think we are using 'flavor' instead of 'branded' in our code, could you double check?

Reply:
When we introduce `flavor`, in order to reduce the impact in our code, we still use this as internal implementation.
I think there is no point of changing our internal implementation.
The value of `IsBranded` is initialized as true if `flavor` is azure, otherwise false.

Reply:
If we want to use `flavor`, it will look like we are using
```
Configuration.Flavor == "azure"
```
everywhere. We should avoid hard-coded strings.
And to simplify, we need to introduce a new property, such as `IsBranded` on `Configuration`, so that we could use just
```
Configuration.IsBranded
```

Reply:
Also considering the existence of MGC, we definitely do not need to change it. It will automatically go away, because the non-azure plugin will take the false part, and our azure plugin takes the true part. There will no longer be a `if`